src/fs.h

```diff
@@ -38,6 +42,41 @@ namespace fsbridge {
         void* hFile = (void*)-1; // INVALID_HANDLE_VALUE
 #endif
     };
+
+
+#if defined WIN32 && defined __GNUC__
```

Comment:
Should this be checking for `__GLIBCXX__` instead of `__GNUC__`? It looks like `__gnu_cxx::stdio_filebuf` might be present in libstdc++ but not libc++ (https://stackoverflow.com/questions/22624503/how-to-get-a-file-descriptor-from-a-stdbasic-ios-for-clang-on-os-x).

Will different workarounds be needed for MSVC and clang?